README.md

```diff
@@ -56,6 +60,70 @@ const output = fragmentImport(fileContents, {
 output; // contains the grapqhl file, with all imports having been inlined
 ```
 
+In the code example above, `fileContents` is the source code of a .graphql file. The return value is graphql source code
+with all the fragment definitions in it from all the various files imported.
+
+### Helper function for parsing imports for your eslint rule
+You can use the function exported from `@graphql-fragment-import/lib/gather-fragment-imports` find and parse all
+fragment definitions into FragmentDefinition nodes (for eslint) by import line number.
+
+```javascript
+const gatherFragmentImports = require('@graphql-fragment-import/lib/gather-fragment-imports');
+
+{
+  // eslint rule
+  create(context)
+  {
+    return {
+      'Document:exit'() {
+        let resolveImport;
+
+        let importResolver =
+          Array.isArray(context.options) &&
+          context.options.length > 0 &&
+          context.options[0].importResolver;
+        if (typeof importResolver === 'string') {
+          if (!path.isAbsolute(importResolver)) {
+            throw new Error(
+              `${ERR_PREFIX} option "importResolver" must be an absolute path, not "${importResolver}"`,
+            );
+          }
+
+          resolveImport = require(importResolver);
+        } else {
+          resolveImport = require('resolve').sync;
+        }
+
+        /**
+         * lineToFragmentNameToBucket is an object whose keys are line numbers and values are objects whose keys are
+         * the fragment names and the parsed eslint node:
+         * {
+         *     1: {
+         *         FooBar: FragmentDefinition,
+         *         Bar: FragmentDefinition
+         *     }
+         * }
+         */
+        const lineToFragmentNameToBucket = gatherFragmentImports({
+          source: context.getSourceCode().text,
+          sourceLocation: filename,
+          resolveImport,  // We recommend using `ember-cli-addon-aware-resolver`
+          fragmentParserGenerator: context.parserServices.getFragmentDefinitionsFromSource, // This should be injected when setting `@eslint-ast/eslint-plugin-graphql/parser` as parser in .eslintrc
+          throwIfImportNotFound: false,
+        });
```

Comment:
When used in eslint rules specifically, three of the arguments all come from the rule context.

What do you think about adding another utility in the eslint-rule package that wraps `gatherFragmentImports` but takes only `context`, `resolveImport` and `throwIfImportNotFound`?

Reply:
great idea